apps/web/utils/firebase.serverside.ts

```diff
@@ -58,13 +76,59 @@ export async function sendRequest(
   }
 
   try {
+    if (await addressCanBeElevatedToTrusted(beneficiary)) {
+      authLevel = AuthLevel.authenticated
+    }
     const db = await getDB()
+    const redis = Redis.fromEnv()
+    const namespace = 'rate-limits'
+    const pendingRequestsInTimePeriod = await redis.keys(
+      `${namespace}:${beneficiary}:*`,
+    )
+
+    if (pendingRequestsInTimePeriod.length >= RATE_LIMITS[authLevel].amount) {
+      return null
+    }
+
     const ref: firebase.database.Reference = await db
       .ref(`${network}/requests`)
       .push(newRequest)
+
+    await redis.set(`${namespace}:${beneficiary}:${ref.key}`, 1, {
+      ex: RATE_LIMITS[authLevel].timePeriodInSeconds,
+    })
+
     return ref.key
   } catch (e) {
     console.error(`Error while sendRequest: ${e}`)
     throw e
   }
 }
+
+const ethPublicClient = createPublicClient({
+  transport: http(),
+  chain: mainnet,
+})
+const celoPublicClient = createPublicClient({
+  transport: http(),
+  chain: celo,
+})
+const LOCKED_CELO_CONTRACT_ADDRESS =
+  '0x6cC083Aed9e3ebe302A6336dBC7c921C9f03349E'
+const WEI = BigInt('1000000000000000000')
+const MIN_ETH_ON_MAINNET = (BigInt(1) * WEI) / BigInt(10) // 0.1 ETH
```

Comment:
hmm actually lets decrease this to like 0.01 ETH i think requiring almost 500$ might be a bit much